sympy/core/basic.py

```diff
@@ -1284,8 +1311,26 @@ def has_free(self, *patterns):
         True
         >>> (x + y + 1).has_free(y + 1)
         True
-
         """
+        if not patterns:
+            return False
+        p0 = patterns[0]
+        if len(patterns) == 1 and iterable(p0) and not isinstance(p0, Basic):
+            # Basic can contain iterables (though not non-Basic, ideally)
+            # but don't encourage mixed passing patterns
+            raise ValueError(filldedent('''
+                Expecting 1 or more Basic args, not a single
+                non-Basic iterable. Don't forget to unpack
+                iterables: `eq.has_free(*patterns)`'''))
+        # try quick test first
+        try:
+            s = set(patterns)
+            rv = self.has_xfree(s)
+            if rv:
+                return rv
+        except TypeError:
```

Comment:
It's usually a bad idea to catch `TypeError`. Why is it being caught here?

Reply:
In case of an un unhashable pattern which is not handled until we get to `_has`.

Reply:
What is an example of an unhashable pattern? Would they not be hashable if sympified?

Reply:
I would still prefer not to catch `TypeError`. It is not a good class of exceptions to catch. At least if we are catching it then the `try` block needs to be restricted in scope to a single operation that is the one where `TypeError` is expected and there should be comments explaining precisely why `TypeError` is being caused.